File: wlr/OGUR/OGUR/GameObjects/Upstairs.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using OGUR.Collision;
using OGUR.Sprites;
using OGUR.Management;
using OGUR.Dungeons;
using OGUR.Creatures;

namespace OGUR.GameObjects
{
    public class Upstairs : GameplayObject
    {
        public Upstairs(Point2 location)
        {
            Initialize(location, SpriteType.UPSTAIRS, GameObjectType.UPSTAIRS,Depth.Stairs);
        }
        public Location GetTargetLocation()
        {
            return Location.Depths;
        }

        ICreature player;
        public override void  Update()
        {
            player = GameplayObjectManager.GetTouchingPlayer(this);
            if(player!=null)
            {
                if (player.IsInteracting())
                {
                    player.PerformInteraction();
                    DungeonFactory.GetPreviousFloor(GetTargetLocation());
                }    
            }
        }
    }
}

File: port-workspace/aigilas-ported/src/com/aigilas/entities/Upstairs.java
package com.aigilas.entities;import com.xna.wrapper.*;import java.util.*;import com.aigilas.creatures.*;import com.aigilas.dungeons.*;import com.spx.entities.*;import com.aigilas.management.*;import com.spx.core.*;
    public class Upstairs  extends  Entity
    {
        public Upstairs(Point2 location)
        {
            Initialize(location, SpriteType.UPSTAIRS, com.aigilas.EntityType.UPSTAIRS,com.aigilas.Depth.Stairs);
        }

        ICreature player;
@Override        public  void  Update()
        {
            player = (Player)EntityManager.GetTouchingCreature(this);
            if(player!=null)
            {
                if (player.IsInteracting())
                {
                    player.PerformInteraction();
                    try {						DungeonFactory.GetPreviousFloor();					} catch (Exception e) {						e.printStackTrace();					}
                }    
            }
        }
    }
